runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -282,6 +283,14 @@ public DirectPipelineResult run(Pipeline pipeline) {
     Collection<ModelEnforcementFactory> parDoEnforcements = createParDoEnforcements(options);
     enforcements.put(ParDo.Bound.class, parDoEnforcements);
     enforcements.put(ParDo.BoundMulti.class, parDoEnforcements);
+    if (options.isTestEncodability()) {
+      enforcements.put(
```

Comment:
Yes; other transforms cannot output elements that they do not receive as inputs.